File: Examples/CSharp/ProgrammersGuide/Load-Save-Convert/ReadVisioDiagram.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace CSharp.ProgrammersGuide.Load_Save_Convert
{
    public class ReadVisioDiagram
    {
        public static void Run()
        {
            //ExStart:ReadVisioDiagram
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_LoadSaveConvert();

            // call the diagram constructor to load a VSD stream
            FileStream st = new FileStream(dataDir + "Drawing1.vsdx", FileMode.Open);
            Diagram vsdDiagram = new Diagram(st);
            st.Close();

            //Call the diagram constructor to load a VDX diagram
            Diagram vdxDiagram = new Diagram(dataDir + "ReadVisioDiagram_Out.vdx");

            /*
             * Call diagram constructor to load a VSS stencil
             * providing load file format
            */
            Diagram vssDiagram = new Diagram(dataDir + "ReadVisioDiagram_Out.vss", LoadFileFormat.VSS);

            /*
             * Call diagram constructor to load diagram from a VSX file
             * providing load options
            */
            LoadOptions loadOptions = new LoadOptions(LoadFileFormat.VSX);
            Diagram vsxDiagram = new Diagram(dataDir + "ReadVisioDiagram_Out.vsx", loadOptions);
            //ExEnd:ReadVisioDiagram
        }
    }
}

File: Examples/VisualBasic/ProgrammersGuide/Load-Save-Convert/ReadVisioDiagram.vb
Imports VisualBasic
Imports System.IO
Imports Aspose.Diagram
Imports System

Public Class ReadVisioDiagram
    Public Shared Sub Run()
        ' ExStart:ReadVisioDiagram
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_LoadSaveConvert()

        ' call the diagram constructor to load a VSD stream
        Dim st As New FileStream(dataDir & Convert.ToString("Drawing1.vsdx"), FileMode.Open)
        Dim vsdDiagram As New Diagram(st)
        st.Close()

        ' call the diagram constructor to load a VDX diagram
        Dim vdxDiagram As New Diagram(dataDir & Convert.ToString("Drawing1.vdx"))

        ' * Call diagram constructor to load a VSS stencil
        ' * providing load file format
        Dim vssDiagram As New Diagram(dataDir & Convert.ToString("Basic.vss"), LoadFileFormat.VSS)

        '* Call diagram constructor to load diagram from a VSX file
        '* providing load options
        Dim loadOptions As New LoadOptions(LoadFileFormat.VSX)
        Dim vsxDiagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsx"), loadOptions)
        ' ExEnd:ReadVisioDiagram
    End Sub
End Class
